addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -442,6 +443,10 @@ function factory(dependencies) {
          * @param {string} data.uuid
          */
         _handleNotificationPartnerChannel(data) {
+            if (data.info === 'channel_seen')
+            {
+                delete data.id; // id of partner seen info, should not be used as thread id
+            }
```

Comment:
Isn't the issue that we reach here in the first place?
Is `_handleNotificationPartnerChannel` correctly handling the new seen info?
And if so, can't we not set an incorrect id in the first place?
The method is expecting to find a thread that is already existing, which will never be the case if proper id is not provided.

Reply:
I don"t really understand why but when it's a chat, it sends a mail.channel notification "channel_seen" which is correctly handled in the right part of the code but when it's not a chat, you get a res.partner notification which makes the execution flow go there. I struggled a long moment to know what should be done here, I think convertData of thread should not handle this but we can't change the content of the notif as it's used elsewhere... Couldn't we make this notification to be parsed by the same function as mail_channel notifications ?

Reply:
I'll need to check this in more details with fresh mind when I'm done with my current PR. I think this one here is too complex to handle as a quick review.